### PR TITLE
feat(connector): [Zen] add apple pay redirect flow support for zen connector

### PR DESCRIPTION
**Pull request by ArjunKarthik**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Zen supports Apple Pay only throw redirect flow, Which requires new type in WalletData which is apple_pay_redirect in the payment confirm call.

API Contract:
`"payment_method_data": {
        "wallet": {
            "apple_pay_redirect": {}
        }
    }`

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
Tested only create payment for Apple Pay as the payment method is available only in connector prod environment.
Screen Shot 2023-06-08 at 12 47 19 AM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/zen/transformers.rs [hunk 1]
```diff
@@ -46,6 +50,27 @@ pub struct ZenPaymentsRequest {
...
+#[serde(rename_all = "camelCase")]
+pub struct CheckoutRequest {
+    amount: String,
+    currency: enums::Currency,
+    custom_ipn_url: String,
+    items: Vec<ZenItemObject>,
+    merchant_transaction_id: String,
+    signature: Option<Secret<String>>,
+    specified_payment_channel: ZenPaymentChannels,
+    terminal_uuid: String,
```
Comment (jarnura) [inline crates/router/src/connector/zen/transformers.rs:69]: Make this as secret

**Review by jarnura — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/zen/transformers.rs [hunk 2]
```diff
@@ -113,109 +138,254 @@ pub struct ZenItemObject {
...
+    )> for ZenPaymentsRequest
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        value: (
+            &types::PaymentsAuthorizeRouterData,
+            &Box<ApplePayRedirectData>,
+        ),
+    ) -> Result<Self, Self::Error> {
+        let (item, _apple_pay_redirect_data) = value;
```
Comment (jagan-jaya) [inline crates/router/src/connector/zen/transformers.rs:226]:
```suggestion
        (item, _apple_pay_redirect_data): (
            &types::PaymentsAuthorizeRouterData,
            &Box<ApplePayRedirectData>,
        ),
    ) -> Result<Self, Self::Error> {
```

**Review by jarnura — APPROVED** [review]

File: crates/router/src/connector/zen/transformers.rs [hunk 3]
```diff
@@ -113,109 +138,254 @@ pub struct ZenItemObject {
...
+        let amount = utils::to_currency_base_unit(item.request.amount, item.request.currency)?;
+        let connector_meta = item.get_connector_meta()?;
```
Comment (jagan-jaya) [inline crates/router/src/connector/zen/transformers.rs:228]: how are we going to handle this in dashboard? is it already notified to them?

File: crates/router/src/connector/zen/transformers.rs [hunk 4]
```diff
@@ -113,109 +138,257 @@ pub struct ZenItemObject {
...
-            timezone: browser_info.time_zone.to_string(),
-            accept_header: browser_info.accept_header,
-            window_size,
-            user_agent: browser_info.user_agent,
+impl TryFrom<(&types::PaymentsAuthorizeRouterData, &GooglePayWalletData)> for ZenPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        value: (&types::PaymentsAuthorizeRouterData, &GooglePayWalletData),
+    ) -> Result<Self, Self::Error> {
+        let (item, gpay_pay_redirect_data) = value;
```
Comment (jagan-jaya) [inline crates/router/src/connector/zen/transformers.rs:186]:
```suggestion
        (item, gpay_pay_redirect_data): (&types::PaymentsAuthorizeRouterData, &GooglePayWalletData),
    ) -> Result<Self, Self::Error> {
```

File: crates/router/src/connector/zen/transformers.rs [hunk 5]
```diff
@@ -113,109 +138,257 @@ pub struct ZenItemObject {
...
+        let mut checkout_request = CheckoutRequest {
+            merchant_transaction_id: item.attempt_id.clone(),
+            specified_payment_channel: ZenPaymentChannels::PclApplepay,
+            currency: item.request.currency,
+            custom_ipn_url: item.request.get_webhook_url()?,
+            items: get_item_object(item, amount.clone())?,
+            amount,
+            terminal_uuid: Secret::new(terminal_uuid),
+            signature: None,
+            url_redirect: router_return_url,
```
Comment (jagan-jaya) [inline crates/router/src/connector/zen/transformers.rs:250]:
```suggestion
            url_redirect: item.request.get_return_url()?,
```

File: crates/router/src/connector/zen/transformers.rs [hunk 6]
```diff
@@ -113,109 +138,257 @@ pub struct ZenItemObject {
...
+            .change_context(errors::ConnectorError::RequestEncodingFailed)?;
+        let terminal_uuid = session
+            .terminal_uuid
+            .clone()
+            .ok_or(errors::ConnectorError::RequestEncodingFailed)?;
+        let router_return_url = item
+            .request
+            .router_return_url
+            .clone()
+            .ok_or(errors::ConnectorError::RequestEncodingFailed)?;
```
Comment (jagan-jaya) [inline crates/router/src/connector/zen/transformers.rs:240]:
```suggestion
```

**Review by jagan-jaya — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/zen/transformers.rs (hunk 3 above)
Comment (ArjunKarthik) [reply crates/router/src/connector/zen/transformers.rs:228]: Dashboard has to do the necessary changes, Already informed

**Review by jarnura — APPROVED** [review]

**Review by jagan-jaya — APPROVED** [review]

**Pull request merged**